src/EPR.Calculator.Frontend/Controllers/LocalAuthorityUploadFileController.cs

```diff
@@ -73,7 +73,7 @@ private async Task<string> GetViewName(IFormFile fileUpload)
             var localAuthorityDisposalCosts = await CsvFileHelper.PrepareLapcapDataForUpload(fileUpload);
 
             this.ViewData["localAuthorityDisposalCosts"] = localAuthorityDisposalCosts.ToArray();
-            this.TempData["LapcapFileName"] = fileUpload.FileName;
+            this.HttpContext.Session.SetString("LapcapFileName", fileUpload.FileName);
```

Comment:
Please create constants for session variables
Also, you need to clean up all the values in try/catch block. Thanks